digdag-standards/src/test/java/io/digdag/standards/operator/td/TdOperatorFactoryTest.java

```diff
@@ -124,4 +124,38 @@ public void rejectQueryFileOutsideOfProjectPath()
         newOperatorFactory(TdOperatorFactory.class)
             .newOperator(newContext(projectPath, newTaskRequest().withConfig(config)));
     }
+
+    @Test
+    public void rejectResultSettingsWithoutResultConnection()
+            throws Exception
+    {
+        Path projectPath = Paths.get("").normalize().toAbsolutePath();
+
+        Config config = newConfig()
+            .set("database", "testdb")
+            .set("query", "select 1")
+            .set("result_settings", "{\"type\":\"http\"}");
+
+        exception.expectMessage("result_settings is valid only if result_connection is set");
+
+        newOperatorFactory(TdOperatorFactory.class)
+            .newOperator(newContext(projectPath, newTaskRequest().withConfig(config)));
+    }
+
+    @Test
+    public void rejectTooLargeConnectionId()
+            throws Exception
+    {
+        Path projectPath = Paths.get("").normalize().toAbsolutePath();
+
+        Config config = newConfig()
+            .set("database", "testdb")
+            .set("query", "select 1")
+            .set("result_connection", "18446744073709551616");
+
+        exception.expectMessage("for key 'result_connection' but got \"18446744073709551616\"");
```

Comment:
right, I had a problem with testing in fact...
In actual use case, digdag command shows more messages like this:

```
2017-03-07 17:11:42 -0800 [ERROR] (0018@+ab+a): Configuration error at task +ab+a: Expected [simple type, class com.google.common.base.Optional<io.digdag.standards.operator.td.IdOrNameParam>] for key 'result_connection' but got 18446744073709551616 (number) (config)
> Instantiation of [simple type, class io.digdag.standards.operator.td.IdOrNameParam] value failed: Must be id or name: 18446744073709551616 (json mapping)
2017-03-07 17:11:43 -0800 [INFO] (0018@+ab^failure-alert): type: notify
error:
  * +ab+a:
    Expected [simple type, class com.google.common.base.Optional<io.digdag.standards.operator.td.IdOrNameParam>] for key 'result_connection' but got 18446744073709551616 (number) (config)
```

It is great if we can easily tell the cause of type mapping failure to Jackson but it's complicated.